PriceKataCalc/Program.cs

```diff
@@ -59,27 +68,56 @@ public double calcDiscount()
             double myprice;
             if (this.discount.beforeTax)
             {
-            myprice= this.price *this.discount.discountValue;
+                myprice = this.price * this.discount.discountValue;
+                myprice = Limit(myprice);
             }
             else
             {
-            myprice= calcPriceAfterTax() *this.discount.discountValue;
+                myprice = calcPriceAfterTax() * this.discount.discountValue;
+                myprice = Limit(myprice);
+
             }
             return myprice;
         }
+
+        private double Limit(double myprice)
+        {
+            if (!Discount.CAP.isPercentage)
+            {
+                if (myprice > Discount.CAP.FlatCost) myprice = Discount.CAP.FlatCost;
+            }
+            else
+            {
+                if (myprice > Discount.CAP.percentage * this.price) myprice = Discount.CAP.percentage * this.price;
+            }
+
+            return myprice;
+        }
+
         public double calcUPCDiscount()
         {
             double myprice;
             if (priceCalc.keysDiscount[this.UPC].beforeTax)
             {
-                myprice = this.price * this.discount.discountValue;
+                if (Discount.CombiningAddictive)
+                {
+                    myprice = (this.price - this.price * UPCDiscount());
+                    myprice = Limit(myprice);
+                }
+                else
+                {
+                    myprice = (this.price - this.price * UPCDiscount());
+                    myprice = Limit(myprice);
+                }
             }
-             myprice = this.price * UPCDiscount();
+            else
+                myprice = this.price * UPCDiscount();
+                myprice=Limit(myprice);
```

Comment:
Couldn't we apply Limit at the end of the method to the discount without considering how it was calculated ?

Reply:
you mean make it work inside the product?
will that be more readable for the user ?